src/PointwiseFunctions/AnalyticSolutions/Errors.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Variables.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Tags {
+/*!
+ * \brief Prefix indicating the difference between two values, such as betwen a
+ * numeric quantity and the corresponding quantity computed from an analytic
+ * solution.
+ */
+template <typename Tag>
+struct Difference : db::PrefixTag, db::SimpleTag {
+  using type = db::const_item_type<Tag>;
+  using tag = Tag;
+};
+
+/*!
+ * \brief For each `Tag` in `TagsList`, where `TagsList` is a subset of
+ * `AnalyticTagsList`, compute the difference `Tag` - `Tags::Analytic<Tag>`.
+ */
+template <typename TagsList, typename AnalyticTagsList>
+struct DifferenceFromAnalyticCompute
+    : db::add_tag_prefix<::Tags::Difference, ::Tags::Variables<TagsList>>,
+      db::ComputeTag {
+  using argument_tags = tmpl::list<
+      Tags::Variables<TagsList>,
+      db::add_tag_prefix<Tags::Analytic, Tags::Variables<AnalyticTagsList>>>;
+  using return_type = ::Variables<db::wrap_tags_in<Tags::Difference, TagsList>>;
+
+  static constexpr void function(
+      const gsl::not_null<return_type*> error,
+      const ::Variables<TagsList>& numeric,
+      const ::Variables<db::wrap_tags_in<Tags::Analytic, AnalyticTagsList>>&
+          analytic) noexcept {
+    (*error) = return_type{numeric.number_of_grid_points()};
+    tmpl::for_each<TagsList>([&error, &numeric, &analytic](auto tag_v) {
+      using tensor_tag = tmpl::type_from<decltype(tag_v)>;
+      for (size_t i = 0; i < db::item_type<tensor_tag>::size(); ++i) {
+        (get<Tags::Difference<tensor_tag>>(*error))[i] =
+            (get<tensor_tag>(numeric))[i] -
+            (get<Tags::Analytic<tensor_tag>>(analytic))[i];
+      }
+    });
+  }
+
+  using base =
+      db::add_tag_prefix<::Tags::Difference, ::Tags::Variables<TagsList>>;
```

Comment:
Also this should be `AnalyticTagsList`. (for readability I suggest moving this line to the very top of the struct, because it needs to be the same as the struct's base class)

Reply:
I moved this as you suggest, but I don't understand how this should be AnalyticTagsList, because AnalyticTagsList might be a superset of TagsList